pkg/dogstatsd/enrich.go

```diff
@@ -42,19 +43,23 @@ func extractTagsMetadata(tags []string, defaultHostname string, originTags strin
 	origin := ""
 	// We use the UDS socket origin if no origin ID was specify in the tags
 	// or 'dogstatsd_entity_id_precedence' is set to False (default false).
-	if entityIDValue == "" || !entityIDPrecedenceEnabled {
+	if originFromTag == "" || !entityIDPrecedenceEnabled {
 		// Add origin tags only if the entity id tags is not provided
-		origin = originTags
+		origin = originFromUDS
 	}
 
 	k8sOrigin := ""
 	// We set k8sOriginID if the metrics contain a 'dd.internal.entity_id' tag different from 'none'.
-	if entityIDValue != "" && entityIDValue != entityIDIgnoreValue {
+	if originFromTag != "" && originFromTag != entityIDIgnoreValue {
 		// Check if the value is not "none" in order to avoid calling
 		// the tagger for entity that doesn't exist.
 
 		// currently only supported for pods
-		k8sOrigin = kubelet.KubePodTaggerEntityPrefix + entityIDValue
+		k8sOrigin = kubelet.KubePodTaggerEntityPrefix + originFromTag
+	} else if originFromTag == "" && originFromMsg != "" {
+		// originFromMsg is the container id sent by the newer clients.
+		// Opt-in is handled when parsing.
+		k8sOrigin = containers.BuildTaggerEntityName(originFromMsg)
```

Comment:
Perhaps this should be re-named `originFromClient` too, to match how we call this value elsewhere.

Reply:
yes, it was done in the second commit 👍